Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/agenda/AgendaDayFragment.kt

```diff
@@ -193,6 +194,15 @@ class AgendaDayFragment : Fragment(), FlexibleAdapter.OnItemClickListener {
         return true
     }
 
+    private fun listenForSearchQueries() {
+        (activity as? MainActivity)?.searchQuery?.observe(
+            viewLifecycleOwner,
+            androidx.lifecycle.Observer {
+                headerAdapter?.setFilter(it)
+                headerAdapter?.filterItems()
```

Comment:
I think we can solve this problem in two ways:
1. Filter by `""` or empty, that way when you tap the search icon it _should_ trigger an unfilter event?
2. This will require a little more work, but rather than searching the recycler view, build out a filter using `AutoCompleteTextView` which actually will give you search results without mutating the parent list. Tapping on a search result in the autocomplete will bring you to the event's page.

Reply:
Ooo I like the auto complete text view idea. That would require us to write our own overlay search widget though, right? Which I guess isn't all that bad.

Reply:
You should be able to use the built-in AutoCompleteSearchView which has the dropdown dialog stuff. The only thing you'd have to do is build your own Adapter/ViewHolder pattern and attach it to the view (same idea for how the RecyclerView is set up)

As for the filtering, as long as you have all of the data you want to "search" on available via the database, you can do what you did already with Implementing `Filterable` and execute the same filtering i.e. ...

```
 @Override
    public Filter getFilter() {
        return new Filter() {
            @Override
            protected FilterResults performFiltering(final CharSequence constraint) {
                 // do the thing with the stuff 
            } 
       }
}
```

Reply:
Right but I'm thinking more of the auto complete part. The auto complete works well if I have a finite list of names, for example, but in this I could search for anything. 

If I start typing `Kotlin` what should I show in the suggestions list? Any talk titles with Kotlin in it? Or in the description?

... but now I think this is what you mean. I can write a custom adapter that actually shows the sessions as I search, and tapping on a suggestion will just launch the page for that session, not filter the list, unless they still explicitly search for something without tapping a suggestion?

Reply:
Exactly!!! It just takes in user input and returns values in the list that match the criteria.

Reply:
okay cool I think you just blew this idea wide open for me. As far as where this text view lives, should I just build some overlay view that shows when you tap on the search icon? Probably easier than trying to show/hide it within the toolbar?

Reply:
Yeah just build a custom AutoCompleteTextView for when you tap on the search icon....maybe you can make it a little custom full screen dialog that is just the input field at the top? It will take over the screen, but if the user presses back it just dismisses the dialog with the AutoCompleteTextView.

Reply:
Great, thanks for the suggestions. Going to close this PR and will reopen when I've made some changes.